Assets/HoloToolkit/Input/Scripts/InputSources/BaseSingletonInputSource.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
I don't think we should make a second singleton class. I'm a little concerned about duplicating code here, and making GazeManager more complicated than it needs to be. I am not a fan of making GazeManager into an input source.